Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5879,10 +5877,23 @@ void WebPageProxy::preferencesDidChange()
 
     // Preferences need to be updated during synchronous printing to make "print backgrounds" preference work when toggled from a print dialog checkbox.
     forEachWebContentProcess([&](auto& webProcess, auto pageID) {
+        std::optional<uint64_t> sharedPreferencesVersion;
+        if (auto sharedPreferences = webProcess.updateSharedPreferencesForWebProcess(preferences())) {
+            sharedPreferencesVersion = sharedPreferences->version;
+#if ENABLE(GPU_PROCESS)
+            if (RefPtr gpuProcess = webProcess.processPool().gpuProcess()) {
+                gpuProcess->sharedPreferencesForWebProcessDidChange(webProcess, *sharedPreferences, [weakWebProcess = WeakPtr { webProcess }, syncedVersion = sharedPreferences->version]() {
+                    if (RefPtr webProcess = weakWebProcess.get())
+                        webProcess->didSyncSharedPreferencesForWebProcess(syncedVersion);
+                });
+            }
+#endif
+        }
+
         if (m_isPerformingDOMPrintOperation)
-            webProcess.send(Messages::WebPage::PreferencesDidChangeDuringDOMPrintOperation(preferencesStore()), pageID,  IPC::SendOption::DispatchMessageEvenWhenWaitingForUnboundedSyncReply);
+            webProcess.send(Messages::WebPage::PreferencesDidChangeDuringDOMPrintOperation(preferencesStore(), sharedPreferencesVersion), pageID,  { IPC::SendOption::IgnoreFullySynchronousMode, IPC::SendOption::DispatchMessageEvenWhenWaitingForUnboundedSyncReply });
```

Comment:
Would be good to explain these IgnoreFullySynchronousMode / IgnoreFullySynchronousMode changes in the commit log. They're not-obvious to me at least.

Reply:
Oh, this change is no longer needed now that I'm not including setPrintBackgrounds as one of preferences. Will revert.

Reply:
Great, that's what I was hoping.